test/rosenbrock.jl

```diff
@@ -121,7 +121,21 @@ sol = solve(prob, NLopt.Opt(:LD_LBFGS, 2))
 sol = solve(prob, GalacticOptim.MOI.OptimizerWithAttributes(NLopt.Optimizer, "algorithm" => :LD_LBFGS))
 @test 10*sol.minimum < l1
 
-sol = solve(prob, NLopt.Opt(:G_MLSL_LDS, 2), nstart=2, local_method = NLopt.Opt(:LD_LBFGS, 2), maxiters=10000)
+sol = solve(prob, NLopt.Opt(:G_MLSL_LDS, 2), local_method = NLopt.Opt(:LD_LBFGS, 2), maxiters=10000)
+@test 10*sol.minimum < l1
+
+prob = OptimizationProblem(optprob, x0)
+sol = solve(prob, NLO(:LN_BOBYQA))
```

Comment:
Update the docs?

Reply:
Will do the docs later today. BTW in the long term would it make sense to conform the `MOI` interface to the general `GalacticOptim` interface such that e.g. kwargs are arguments to `MOI.OptimiserWithAttributes`. That would make it easier rather than switching between `kwargs` for most optimisers and the `Pairs` approach of MOI or should the MOI interface stay more low level?

Reply:
We want users to be using a common set of kwargs as much as possible. The Pairs MOI thing is just a hack for handling the cases we haven't wrapped.

Reply:
And just to mention that unaccounted for `kwargs` splatting is not the nicest experience for maintainability

Reply:
That makes sense. The unaccounted for `kwargs` arguments as they stand at least allows the user to set all the settings of the  underlying optimiser without us explicitly mapping it. Should each interface maybe have `map_optimiser_args` function which depending on the optimiser is passed to it maps the common set parameters e.g. maxiters, maxtime, ftol, gtol to the respective settings of the given optimiser. While at the same time one can still have the `kwargs...` approach which allows user fine tuning and full access to the underlying interface.

Reply:
yeah, we need to do a map for each wrapper. It just hasn't been done yet, but everything should match the SciML common interface kwargs in the end, with special kwargs in the alg.

Reply:
I might have a go at this over the next week. Is there a documentation of the SciML common interface kwargs somewhere. One could "force" those arguments to all `GalacticOptim` interfaces and let the mapping function handle warning message if a certain common kwargs has not been implemented yet and requires the user defne the optimiser specific kwargs. That would make it also easier to extend in the future and means that the optimiser is not silently ignoring arguments like what was happing in the `NLopt` case. I will stop now and create an issue regarding the common interface later today or tomorrow.